internal/events/event_test.go

```diff
@@ -205,390 +208,132 @@ var _ = Describe("Events library", func() {
-					username:    "admin_user1",
-					context:     context.WithValue(context.Background(), restapi.AuthKey, &ocm.AuthPayload{Username: "admin_user1", Role: ocm.AdminRole}),
-					createHost:  true,
-					querySucess: true,
-					err:         nil,
-				},
-				{
-					name:        "filter by host_id - no user in context",
-					username:    "",
-					context:     context.TODO(),
-					createHost:  true,
-					querySucess: true,
-					err:         nil,
-				},
-				{
-					name:        "filter by an invalid host_id - no user in context",
-					username:    "",
-					context:     context.TODO(),
-					createHost:  false,
-					querySucess: false,
-					err:         gorm.ErrRecordNotFound,
-				},
-				{
-					name:        "filter by an invalid host_id - user with user role",
-					username:    "test_user1",
-					context:     context.WithValue(context.Background(), restapi.AuthKey, &ocm.AuthPayload{Username: "test_user1", Role: ocm.UserRole}),
-					createHost:  false,
-					querySucess: false,
-					err:         gorm.ErrRecordNotFound,
-				},
-			}
-			for i := range tests {
-				test := tests[i]
-				It(test.name, func() {
-					hostId := strfmt.UUID(uuid.New().String())
-					if test.createHost {
-						test_host := &common.Host{
-							KubeKeyNamespace: "test_namespace",
-							Host: models.Host{
-								ID:         &hostId,
-								InfraEnvID: infraEnv1,
-								UserName:   test.username,
-							},
-						}
-						Expect(db.Create(&test_host).Error).ShouldNot(HaveOccurred())
-					}
-					theEvents.V2AddEvent(test.context, nil, &hostId, nil, "fake_event", models.EventSeverityInfo, models.EventCategoryUser, time.Now())
-					evs, err := theEvents.V2GetEvents(test.context, nil, &hostId, nil, models.EventCategoryUser)
-					if test.querySucess {
-						Expect(err).Should(BeNil())
-						Expect(len(evs)).Should(Equal(1))
-					} else {
-						Expect(err).ShouldNot(BeNil())
-						Expect(errors.Is(err, test.err)).Should(Equal(true))
-						Expect(len(evs)).Should(Equal(0))
-					}
-				})
-			}
+		Context("with admin role", func() {
+			BeforeEach(func() {
+				payload := &ocm.AuthPayload{}
+				payload.Role = ocm.AdminRole
+				ctx = context.WithValue(context.TODO(), restapi.AuthKey, payload)
+			})
+			It("gets all events", func() {
+				evs, err := theEvents.V2GetEvents(ctx, nil, nil, nil)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(len(evs)).To(Equal(6))
+			})
+
+			It("gets cluster's events when specifying cluster", func() {
+				evs, err := theEvents.V2GetEvents(ctx, &cluster1, nil, nil)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(len(evs)).To(Equal(2))
+			})
+
+			It("gets infra-env's events when specifying infra-env", func() {
+				evs, err := theEvents.V2GetEvents(ctx, nil, nil, &infraEnv1)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(len(evs)).To(Equal(3))
+			})
+			It("gets hosts's events when specifying host", func() {
```

Comment:
Should we have a test to fetch events by host_id alone?

Reply:
added